### ISSUE TITLE
Another matching issue (Cyclic structures)

### ISSUE THREAD

**Issue by MrDarkHorse** [partridgejiang/Kekule.js#93]
<img width="731" alt="screen shot 2018-12-13 at 1 11 58 pm" src="https://user-images.githubusercontent.com/9271182/49961632-c5d10100-fed8-11e8-9fb9-bdda685e34f5.png">

**Comment by MrDarkHorse** [comment]
I'm increasingly of the opinion that this change likely caused the regression.  I will attempt to prove this tomorrow

https://github.com/partridgejiang/Kekule.js/commit/e1767d7dfa7e935025d56ba09bf35d8ab5bdb01d

**Comment by partridgejiang** [comment]
@MrDarkHorse  Thanks a lot again. A rough fix has been made in commit 8401caa, it seems to solve the problem.

**Comment by MrDarkHorse** [comment]
I don't know.  Just seems like the canonical index isn't getting set correctly.  I get lost in the graph theory stuff though. :\

**Comment by partridgejiang** [comment]
A little bewildered... In my test:
![screen shot 2018-12-19 at 09 14 26](https://user-images.githubusercontent.com/10516825/50192814-e1238c80-036e-11e9-8bfe-da2cdc5d8557.png)
@MrDarkHorse Could you please provide the source files of your test cases? Thanks.

**Comment by MrDarkHorse** [comment]
`{
	"coordPos2D": 0,
	"coordPos3D": 0,
	"renderOptions": {
		"expanded": true,
		"__type__": "object"
	},
	"coord2D": {
		"x": 2.045086640267364,
		"y": 5.58901262913221,
		"__type__": "object"
	},
	"charge": 0,
	"parity": null,
	"ctab": {
		"nodes": [{
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0,
				"y": -0.8000000000000025,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0.6928203230275525,
				"y": -0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": -0.6928203230275543,
				"y": -0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0.6928203230275525,
				"y": 0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": -0.6928203230275534,
				"y": 0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0,
				"y": 0.8000000000000025,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}],
		"anchorNodes": [],
		"connectors": [{
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [0, 1]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [0, 2]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"bondType": "covalent",
			"bondOrder": 2,
			"electronCount": 4,
			"isInAromaticRing": false,
			"connectedObjs": [1, 3]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [2, 4]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [3, 5]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [4, 5]
		}],
		"__type__": "Kekule.StructureConnectionTable"
	},
	"__type__": "Kekule.Molecule"
}`

**Comment by MrDarkHorse** [comment]
`{
	"coordPos2D": 0,
	"coordPos3D": 0,
	"renderOptions": {
		"expanded": true,
		"__type__": "object"
	},
	"coord2D": {
		"x": 2.045086640267364,
		"y": 5.58901262913221,
		"__type__": "object"
	},
	"charge": 0,
	"parity": null,
	"ctab": {
		"nodes": [{
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0,
				"y": -0.8000000000000025,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0.6928203230275525,
				"y": -0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": -0.6928203230275543,
				"y": -0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0.6928203230275525,
				"y": 0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": -0.6928203230275534,
				"y": 0.40000000000000036,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}, {
			"__type__": "Kekule.Atom",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"coord2D": {
				"x": 0,
				"y": 0.8000000000000025,
				"__type__": "object"
			},
			"charge": 0,
			"parity": null,
			"explicitHydrogenCount": 0,
			"isotopeId": "C"
		}],
		"anchorNodes": [],
		"connectors": [{
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [0, 1]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [0, 2]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"bondType": "covalent",
			"bondOrder": 2,
			"electronCount": 4,
			"isInAromaticRing": false,
			"connectedObjs": [1, 3]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [2, 4]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [3, 5]
		}, {
			"__type__": "Kekule.Bond",
			"coordPos2D": 0,
			"coordPos3D": 0,
			"parity": null,
			"bondType": "covalent",
			"bondOrder": 1,
			"electronCount": 2,
			"isInAromaticRing": false,
			"connectedObjs": [4, 5]
		}],
		"__type__": "Kekule.StructureConnectionTable"
	},
	"__type__": "Kekule.Molecule"
}`

**Comment by partridgejiang** [comment]
Still these two structures are same in my test with the latest commits...
![image](https://user-images.githubusercontent.com/10516825/50400067-d3cb2f80-07be-11e9-856d-fc694a9894be.png)

**Comment by MrDarkHorse** [comment]
So I'm running the structure compare demo tool in master, and I pasted the above structures in using the JSON loader.  

I modified the demo file to point to the local src, like so:
```
<title>Molecule Compare</title>

<script src="../../libs/raphael-min.2.0.1.js"></script>
<script src="../../../src/kekule.js"></script>  
```

Still coming up as different.

![screen shot 2019-01-02 at 2 49 39 pm](https://user-images.githubusercontent.com/9271182/50612035-2af79100-0e9e-11e9-9349-96b90f2852d1.png)

Are you sure you're both running the latest code and using the exact JSON files that I'm using?
